snowplow_tracker/test/unit/test_in_memory_event_store.py

```diff
@@ -0,0 +1,101 @@
+# """
+#     test_in_memory_event_store.py
+
+#     Copyright (c) 2013-2022 Snowplow Analytics Ltd. All rights reserved.
+
+#     This program is licensed to you under the Apache License Version 2.0,
+#     and you may not use this file except in compliance with the Apache License
+#     Version 2.0. You may obtain a copy of the Apache License Version 2.0 at
+#     http://www.apache.org/licenses/LICENSE-2.0.
+
+#     Unless required by applicable law or agreed to in writing,
+#     software distributed under the Apache License Version 2.0 is distributed on
+#     an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+#     express or implied. See the Apache License Version 2.0 for the specific
+#     language governing permissions and limitations there under.
+
+#     Authors: Anuj More, Alex Dean, Fred Blundun, Paul Boocock
+#     Copyright: Copyright (c) 2013-2022 Snowplow Analytics Ltd
+#     License: Apache License Version 2.0
+# """
+
+import unittest
+from snowplow_tracker.event_store import InMemoryEventStore
+
+
+class TestInMemoryEventStore(unittest.TestCase):
+    def setUp(self) -> None:
+        pass
+
+    def test_init(self):
+        event_store = InMemoryEventStore()
+        self.assertEqual(event_store.buffer_capacity, 10000)
+        self.assertEqual(event_store.event_buffer, [])
+
+    def test_add_event(self):
+        event_store = InMemoryEventStore()
+        nvPairs = {"n0": "v0", "n1": "v1"}
+
+        event_store.add_event(nvPairs)
+        self.assertDictEqual(nvPairs, event_store.event_buffer[0])
+
+    def test_size(self):
+        event_store = InMemoryEventStore()
+        nvPairs = {"n0": "v0", "n1": "v1"}
+
+        event_store.add_event(nvPairs)
+        event_store.add_event(nvPairs)
+        event_store.add_event(nvPairs)
+
+        self.assertEqual(event_store.size(), 3)
+
+    def add_failed_events_to_buffer(self):
+        event_store = InMemoryEventStore()
+
+        nvPairs = {"n0": "v0", "n1": "v1"}
+
+        event_store.add_event(nvPairs)
+        event_store.add_event(nvPairs)
+        payload_list = event_store.event_buffer
+        event_store.cleanup(payload_list, True)
+
+        self.assertEqual(event_store.event_buffer, payload_list)
+
+    def remove_success_events_from_buffer(self):
+        event_store = InMemoryEventStore()
+
+        nvPairs = {"n0": "v0", "n1": "v1"}
+
+        event_store.add_event(nvPairs)
+        event_store.add_event(nvPairs)
+        payload_list = event_store.event_buffer
+        event_store.cleanup(payload_list, False)
+
+        self.assertEqual(event_store.event_buffer, [])
+
+    def drop_new_events_buffer_full(self):
```

Comment:
You need to prepend `test_` to the name of the function for pytest to recognize and run the test.

Reply:
This test actually fails which is a good point since we don't check the buffer capacity in the `add_event` function – we should probably do that and log a message with `logger.error()` in case the event doesn't fit into the capacity.